contracts/bonding/BondingManager.sol

```diff
@@ -1029,12 +1029,13 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             // we know they will require too much gas to loop through all the necessary rounds to claim earnings
             // The user should instead manually invoke `claimEarnings` to split up the claiming process
             // across multiple transactions
-            require(_endRound.sub(del.lastClaimRound) <= maxEarningsClaimsRounds);
+            uint256 lastClaimRound = del.lastClaimRound;
```

Comment:
Can we instead store `uint256 startRound = del.lastClaimRound + 1` so that we can use `startRound` in the for loop conditional as well as when we emit the `EarningsClaimed` event? We would just need to change the below require statement to:

```
require(_endRound.sub(startRound) < maxEarningsClaimsRounds);
```

Reply:
We can do that but `_endRound.sub(startRound)` will revert when we invoke `autoClaimEarnings` twice in the same round so we'll still have to use `_endRound.sub(del.lastClaimedRound)`

Reply:
Great point!

Can we avoid an extra storage read for `del.lastClaimRound` by storing `lastClaimRound` in memory?

Ex.

```
uint256 lastClaimRound = del.lastClaimRound;
require(_endRound.sub(lastClaimRound) <= maxEarningsClaimsRounds);

uint256 startRound = lastClaimRound.add(1);

// Rest of code
```

Reply:
With some work-arounds yes, otherwise we'll be running into the stack size limit and we'll pop off the `_delegator` argument , which becomes problematic if we need to access that value on the stack down the line (the notorious "stack too deep" error). 

I think removing `isTranscoder` should work, we can evaluate that expression directly instead.

It's for the same reason (stack size limit) that we are doing arithmetics in the line where the event gets logged. The obvious solution would be to keep track of a local `totalFees` and `totalRewards` variable, but due EVM restrictions it's not possible.

Reply:
Fixed in 9f4751a